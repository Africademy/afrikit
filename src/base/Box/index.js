// @flow
import styled from 'styled-components';
import {
  space,
  width,
  flex,
  color,
  fontSize,
  order,
  alignSelf,
  theme,
} from '../../theme';
import { cleanTag } from '../../utils';

const Box = styled(cleanTag)(
  [],
  { boxSizing: 'border-box' },
  width,
  space,
  fontSize,
  color,
  flex,
  order,
  alignSelf,
);

Box.displayName = 'Box';

Box.defaultProps = {
  theme,
};

Box.propTypes = {
  ...width.propTypes,
  ...space.propTypes,
  ...fontSize.propTypes,
  ...color.propTypes,
  ...flex.propTypes,
  ...order.propTypes,
  ...alignSelf.propTypes,
};

export default Box;
