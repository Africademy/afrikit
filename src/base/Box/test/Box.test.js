// @flow
import React from 'react';
import expect from 'expect';
import { create as render } from 'react-test-renderer';
import Box from '../index';

describe('Box Component', () => {
  test('Box renders', () => {
    const json = render(<Box m={2} px={3} />).toJSON();
    expect(json).toMatchSnapshot();
  });

  test('Box renders with props', () => {
    const json = render(
      <Box
        m={[1, 2]}
        px={[1, 2]}
        w={1}
        flex="1 1 auto"
        alignSelf="flex-start"
      />,
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  // clean
  test('Box removes grid-styled props', () => {
    const json = render(
      React.createElement(Box, {
        id: 'hi',
        width: 0.5,
        color: 'blue',
        fontSize: 4,
        wrap: true,
        flexWrap: 'wrap',
      }),
    ).toJSON();
    expect(json.props.id).toEqual('hi');
    expect(json.props.width).toEqual(undefined);
    expect(json.props.color).toEqual(undefined);
    expect(json.props.fontSize).toEqual(undefined);
    expect(json.props.wrap).toEqual(undefined);
  });

  test('Box accepts an is prop to change elements', () => {
    const json = render(
      React.createElement(Box, {
        is: 'h2',
      }),
    ).toJSON();
    expect(json.type).toEqual('h2');
  });
});
