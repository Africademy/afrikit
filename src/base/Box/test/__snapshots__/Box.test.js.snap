// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component Box renders 1`] = `
<div
  className="sc-bdVaJa goNBYt"
/>
`;

exports[`Box Component Box renders with props 1`] = `
<div
  className="sc-bdVaJa chFJb"
/>
`;
